scoap3/cli_fixes.py

```diff
@@ -252,3 +253,50 @@ def fix_publication_date(dry_run, ids, min_day_diff):
 
     if dry_run:
         error('NO CHANGES were committed to the database, because --dry-run flag was present.')
+
+
+def change_oai_hostname_for_record(record, old_hostname, new_hostname, dry_run):
+    if not record.json:
+        rerror('no json', record)
+        return
+
+    if 'id' not in record.json.get('_oai', {}):
+        rerror('no oai information', record)
+        return
+
+    pid = PersistentIdentifier.query\
+        .filter(PersistentIdentifier.object_uuid == record.id)\
+        .filter(PersistentIdentifier.pid_type == 'oai').one()
```

Comment:
Do you expect at this stage to always get a PID? If not, then would be safer to use `one_or_none()` and check if it's not None

Reply:
Actually we should always have one, but you are definitely right, especially as I think it's not the case (but something we'll have to fix).